File: src/main/java/com/linebot/action/ActionSelector.java
package com.linebot.action;

import lombok.AllArgsConstructor;
import lombok.Getter;

import javax.annotation.Nullable;
import javax.validation.constraints.NotNull;
import java.util.Arrays;
import java.util.List;

@AllArgsConstructor
public enum ActionSelector {
    TRAIN_DELAY("電車遅延設定", Arrays.asList("trainDelayInputAction", "TrainDelayAction")),
    TOBUY_ADD("買い物リスト追加", Arrays.asList("tobuyAddInputAction", "tobuyAddAction")),
    TOBUY_CONFIRM("買い物リスト確認", Arrays.asList("tobuyConfirmAction")),
    TOBUY_COMPLATE("買い物リスト購入", Arrays.asList("tobuyComplateInputAction", "tobuyAddAction")),
    QIITA_SETTING("qiita設定", Arrays.asList("qiitaIdInputAction", "qiitaIdAction")),
    QIITA_ITEM_LIST("qiita記事一覧取得", Arrays.asList("qiitaItemAction")),
    ORDER("要望・バグ", Arrays.asList("orderInputAction", "orderAction"));

    private String startWord;
    private List<String> actionList;

    // kotlin 用にgetterを用意する
    public String getStartWord() {
        return this.startWord;
    }

    public List<String> getActionList() {
        return this.actionList;
    }

    @Nullable
    public static ActionSelector getByStartWord(@NotNull String message) {
        for (ActionSelector action : values()) {
            if (action.getStartWord().equals(message)) {
                return action;
            }
        }
        return null;
    }

}

File: src/main/kotlin/com/linebot/action/ActionSelector.kt
package com.linebot.action

enum class ActionSelector(
    val startWord: String,
    val actionList: List<String>

) {
    TOBUY_ADD("買い物リスト追加", listOf("tobuyAddInputAction", "tobuyAddAction")),
    TOBUY_CONFIRM("買い物リスト確認", listOf("tobuyConfirmAction")),
    TOBUY_COMPLATE("買い物リスト購入", listOf("tobuyComplateInputAction", "tobuyAddAction"));
    companion object {
        fun getByStartWord(message: String): ActionSelector? = values().firstOrNull { it.startWord == message }
    }
}
